doc/book/box/indexes.rst

```diff
@@ -28,49 +27,225 @@ Indexes have certain limitations. See details on page
+    put it in a variable ``my_space``:
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> my_space = box.schema.space.create('tester')
+
+#.  Format the created space by specifying field names and types:
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> my_space:format({
+                 > {name = 'id', type = 'unsigned'},
+                 > {name = 'band_name', type = 'string'},
+                 > {name = 'year', type = 'unsigned'},
+                 > {name = 'rate', type = 'unsigned', is_nullable=true}})
+
+#.  Create the **primary** index (named ``primary``):
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> my_space:create_index('primary', {
+                 > type = 'hash',
+                 > parts = {'id'}
+                 > })
+
+    This is a primary index based on the ``id`` field of each tuple.
+
+#.  Insert some :ref:`tuples <index-box_tuple>` into the space:
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> my_space:insert{1, 'Roxette', 1986, 1}
+        tarantool> my_space:insert{2, 'Scorpions', 2015, 4}
+        tarantool> my_space:insert{3, 'Ace of Base', 1993}
+        tarantool> my_space:insert{4, 'Roxette', 2016, 3}
+
+#.  Create a **secondary index**:
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> box.space.tester:create_index('secondary', {parts = {{field=3, type='unsigned'}}})
+        ---
+        - unique: true
+          parts:
+          - type: unsigned
+            is_nullable: false
+            fieldno: 3
+          id: 2
+          space_id: 512
+          type: TREE
+          name: secondary
+        ...
+
+#.  Create a **multi-part index** with three parts:
+
+    ..  code-block:: tarantoolsession
+
+        tarantool> box.space.tester:create_index('thrine', {parts = {
+                 > {field = 2, type = 'string'},
+                 > {field=3, type='unsigned'},
+                 > {field=4, type='unsigned'}
+                 > }})
+        ---
+        - unique: true
+          parts:
+          - type: string
+            is_nullable: false
+            fieldno: 2
+          - type: unsigned
+            is_nullable: false
+            fieldno: 3
+          - type: unsigned
+            is_nullable: true
+            fieldno: 4
+          id: 6
+          space_id: 513
+          type: TREE
+          name: thrine
+        ...
+
+**There are the following SELECT variations:**
+
+* The search can use **comparisons** other than equality:
+
+  ..  code-block:: tarantoolsession
+
+      tarantool> box.space.tester:select(1, {iterator = 'GT'})
+      ---
+      - - [2, 'Scorpions', 2015, 4]
+        - [3, 'Ace of Base', 1993]
+        - [4, 'Roxette', 2016, 3]
+      ...
+
+  The :ref:`comparison operators <box_index-iterator-types>` are:
+
+  *   ``LT`` for "less than"
+  *   ``LE`` for "less than or equal"
+  *   ``GT`` for "greater"
+  *   ``GE`` for "greater than or equal" .
+  *   ``EQ`` for "equal",
+  *   ``REQ`` for "reversed equal"
```

Comment:
It's relevant only for tree index. RTREE and BITSET have their own iterator types.
HASH index has such iterators but they have a bit different semantic (compare hashes AFAIK).